review_app_status.py

```diff
@@ -133,19 +136,28 @@ def _get_build_data(url: str, interval: int) -> dict:
 
 
 def _check_review_app_deployment_status(
-    review_app_url: str, accepted_responses: t.List[int]
+    review_app_url: str, accepted_responses: t.List[int], timeout: int, interval: int
 ):
     """Check Review App deployment status code against accepted_responses.
     
     Inputs:
         review_app_url: URL of the Review App to be checked.
-        accepted_responses: status codes to be accepted.
+        accepted_responses: Status codes to be accepted.
+        timeout: Maximum time to spend retrying the HTTP response check until it succeeds.
+        interval: Interval for each HTTP response check.
     """
-    time.sleep(5)  # Let the deployment breathe.
-    r = requests.get(review_app_url)
-    review_app_status = r.status_code
-    logger.info(f"Review app status: {review_app_status}")
-    if review_app_status not in accepted_responses:
+    if interval > timeout:
+        raise ValueError("Interval can't be greater than publish_timeout.")
+
+    while timeout > 0:
+        r = requests.get(review_app_url)
+        review_app_status = r.status_code
+        logger.info(f"Review app status: {review_app_status}")
+        if review_app_status in accepted_responses:
+            return
+        time.sleep(interval)
+        timeout -= interval
+
         r.raise_for_status()
```

Comment:
This should be outside of the loop.

```suggestion
    r.raise_for_status()
```

Reply:
I think it's not required.

Reply:
Removed the line

Reply:
It is required, We aren't specifying any action after the timeout, it will pass quietly after the loop.  An error is required if the application returns a non-accepted-response after the timeout.

Reply:
Right it's required. But instead of `r.raise_for_status()` I think it's better to raise the error manually since we are specifying the response to expect explicitly. It will then raise error even when the response is correct, but we aren't expecting that.